pyomo/contrib/pyros/pyros.py

```diff
@@ -210,6 +219,49 @@ def _log_disclaimer(self, logger, **log_kwargs):
         logger.log(msg="https://github.com/Pyomo/pyomo/issues/new/choose", **log_kwargs)
         logger.log(msg="=" * self._LOG_LINE_LENGTH, **log_kwargs)
 
+    def _log_config_user_values(
+        self, logger, config, exclude_options=None, **log_kwargs
+    ):
+        """
+        Log explicitly set PyROS solver options.
+
+        If there are no such options, or all such options
+        are to be excluded from consideration, then nothing is logged.
+
+        Parameters
+        ----------
+        logger : logging.Logger
+            Logger for the solver options.
+        config : ConfigDict
+            PyROS solver options.
+        exclude_options : None or iterable of str, optional
+            Options (keys of the ConfigDict) to exclude from
+            logging. If `None` passed, then the names of the
+            required arguments to ``self.solve()`` are skipped.
+        **log_kwargs : dict, optional
+            Keyword arguments to each statement of ``logger.log()``.
+        """
+        if exclude_options is None:
+            exclude_options = set(self._DEFAULT_CONFIG_USER_OPTIONS)
+        else:
+            exclude_options = set(exclude_options)
+
+        user_values = list(
+            filter(
+                # note: first clause of logical expression
+                #       accounts for bug(?) causing an iterate
+                #       of user_values to be the config dict itself
```

Comment:
Is this related to #3721? If so, it might be worthwhile to put that issue number in here so people can go read more about it.